qt/python/mantidqt/project/recovery/recoverygui/recoveryfailureview.py

```diff
@@ -0,0 +1,96 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+
+from qtpy.QtWidgets import QDialog, QHeaderView, QTableWidgetItem
+from qtpy.QtCore import Signal, Slot, Qt
+from mantidqt.utils.qt import load_ui
+from mantid.kernel import logger
+
+
+class RecoveryFailureView(QDialog):
+    def __init__(self, presenter):
+        super(RecoveryFailureView, self).__init__()
+        self.ui = load_ui(__file__, "RecoveryFailure.ui", baseinstance=self)
+        self.ui.tableWidget.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+        self.ui.tableWidget.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+        self.presenter = presenter
+
+        # Make sure the UI doesn't leak memory
+        self.ui.setAttribute(Qt.WA_DeleteOnClose, True)
+
+        # Set the table data
+        self._add_data_to_table()
+
+    def _add_data_to_table(self):
+        # This table's size was generated for 5 which is the default but will take the
+        number_of_rows = self.presenter.get_number_of_checkpoints()
+        for ii in range(0, number_of_rows):
+            row = self.presenter.get_row(ii)
+            for jj in range(0, len(row)):
+                self.ui.tableWidget.setItem(ii, jj, QTableWidgetItem(row[jj]))
+
+    def reject(self):
+        self.presenter.start_mantid_normally()
+
+    def set_progress_bar_maximum(self, new_value):
+        self.ui.progressBar.setMaximum(new_value)
+
+    def connect_progress_bar(self):
+        self.presenter.project_recovery.multi_file_interpreter.current_editor().sig_progress.connect(
+            self.update_progress_bar)
+
+    def emit_abort_script(self):
+        self.abort_project_recovery_script.connect(self.presenter.project_recovery.multi_file_interpreter.abort_all)
+        logger.error("Project Recovery: Cancelling recovery")
+        self.abort_project_recovery_script.emit()
+
+    def change_start_mantid_button(self, string):
+        self.ui.pushButton_3.setText(string)
+
+    ######################################################
```

Comment:
On second thought, these sections should be removed, they are accurate now, but over time, if new functions are added they might become misleading.

Reply:
They should be added to the correct sections or their PR should be rejected. They are present for clarity and for help navigating a large class, I do not wish to sacrifice that as I use them and other devs may do, when working in this class.

Reply:
After a discussion we have resolved this and I am happy for them to be left in